libkbfs/tlf_journal.go

```diff
@@ -516,17 +519,19 @@ func (j *tlfJournal) flushBlockEntries(
 }
 
 func (j *tlfJournal) getNextMDEntryToFlush(ctx context.Context,
-	currentUID keybase1.UID, currentVerifyingKey VerifyingKey) (
+	currentUID keybase1.UID, currentVerifyingKey VerifyingKey,
+	end MetadataRevision) (
 	MdID, *RootMetadataSigned, error) {
 	j.journalLock.RLock()
 	defer j.journalLock.RUnlock()
 	return j.mdJournal.getNextEntryToFlush(
-		ctx, currentUID, currentVerifyingKey, j.config.Crypto())
+		ctx, currentUID, currentVerifyingKey, end, j.config.Crypto())
 }
 
 func (j *tlfJournal) convertMDsToBranch(
 	ctx context.Context, currentUID keybase1.UID,
-	currentVerifyingKey VerifyingKey) (MdID, *RootMetadataSigned, error) {
+	currentVerifyingKey VerifyingKey,
+	end MetadataRevision) (MdID, *RootMetadataSigned, error) {
```

Comment:
This confused me at first because I thought you were going to use it to only convert the journal up to the `end` entry.  But you are just using it to get the next entry to flush.  Not sure how to fix that -- maybe call this `nextExtryEnd` or something?